crates/driver/src/infra/simulator/mod.rs

```diff
@@ -114,11 +129,28 @@ enum Inner {
 }
 
 #[derive(Debug, thiserror::Error)]
-pub enum Error {
+#[error("err: {err}, tx: {tx:?}")]
+pub struct Error {
+    err: InnerError,
+    tx: eth::Tx,
```

Comment:
Probably doesn't make a huge difference, but I'd find it slightly cleaner if the tx was only on `driver::domain::competition::solution::Error::Simulation` (since here you are gossiping back the tx to the caller that already has passed it in)

Reply:
Hm, probably wont end up cleaner since I need to make `with_tx` available to callsite which seems like unnecessary leaking all the errors and conversion from `infra::simulator`. I like to keep it all inside `simulator`, especially since I changed the `tx` to be optional. We don't want to carry and log `tx` for http connection errors for example.

Reply:
You wouldn't be leaking any error types though would you? `with_tx` would simply be defined in the domain. Your Simulation `solution::Error::Simulation` would then just have an inner simulation error field (which it already has) and a tx. But sure doesn't make a big difference.

Reply:
Agreed. Fixed.